ledger-core/v2/virtual/object/pending_table_test.go

```diff
@@ -54,27 +54,40 @@ func TestPendingList(t *testing.T) {
 	pl := NewRequestList()
 	require.Equal(t, 0, pl.Count())
 	require.Equal(t, 0, pl.CountFinish())
-	require.Equal(t, pulse.Number(0), pl.earliestPulse)
+	require.Equal(t, pulse.Number(0), pl.EarliestPulse())
 
 	require.Equal(t, true, pl.Add(RefOne))
+	require.Equal(t, true, pl.Exist(RefOne))
 	require.Equal(t, 1, pl.Count())
-	require.Equal(t, nextPulseNumber, pl.earliestPulse)
+	require.Equal(t, nextPulseNumber, pl.EarliestPulse())
 
 	require.Equal(t, false, pl.Add(RefOne))
+	require.Equal(t, true, pl.Exist(RefOne))
 	require.Equal(t, 1, pl.Count())
-	require.Equal(t, nextPulseNumber, pl.earliestPulse)
+	require.Equal(t, nextPulseNumber, pl.EarliestPulse())
 
 	require.Equal(t, true, pl.Add(RefOld))
+	require.Equal(t, true, pl.Exist(RefOne))
+	require.Equal(t, true, pl.Exist(RefOld))
 	require.Equal(t, 2, pl.Count())
-	require.Equal(t, pd.PulseNumber, pl.earliestPulse)
+	require.Equal(t, pd.PulseNumber, pl.EarliestPulse())
 
 	require.Equal(t, true, pl.Add(RefTwo))
+	require.Equal(t, true, pl.Exist(RefOne))
+	require.Equal(t, true, pl.Exist(RefOld))
+	require.Equal(t, true, pl.Exist(RefTwo))
 	require.Equal(t, 3, pl.Count())
-	require.Equal(t, pd.PulseNumber, pl.earliestPulse)
+	require.Equal(t, pd.PulseNumber, pl.EarliestPulse())
 	require.Equal(t, 0, pl.CountFinish())
 
-	pl.Finish(RefOld)
+	pl.Finish(RefOne)
+	require.Equal(t, pd.PulseNumber, pl.EarliestPulse()) // doesn't change
 	require.Equal(t, 1, pl.CountFinish())
+	require.Equal(t, 2, pl.CountActive())
+
+
+	nonExistRefFinished := pl.Finish(reference.NewSelf(gen.UniqueIDWithPulse(currentPulse)))
```

Comment:
confusing variable name, `nonExistRefFinished` really is `referenceExists`

Reply:
here I try to call pl.Finish on ref, that wasn't in a list. Renamedd it @bigbes

Reply:
You try to describe in that variable a test case, but i think that it's not right :) No need to do that, just describe what that value is and in commentary describe what that test tests.

Otherwise it looks silly :)
```
tryTooFinishRefThatNotInList := pl.Finish(reference.NewSelf(gen.UniqueIDWithPulse(currentPulse)))
require.Equal(t, false, tryTooFinishRefThatNotInList)
```
why variable `tryTooFinishRefThatNotInList` is false? 
```
// try to finish reference, that not in a list
referenceExists := pl.Finish(reference.NewSelf(gen.UniqueIDWithPulse(currentPulse)))
require.Equal(t, false, referenceExists)
```

because (and i've looked into code) return value can be false ONLY if reference does not exists, and then all makes perfect sense: retvalue is false means reference does not exists

Reply:
fixed